server/src/main/java/org/elasticsearch/search/aggregations/AggregationBuilder.java

```diff
@@ -52,6 +55,10 @@ protected AggregationBuilder(String name) {
         this.name = name;
     }
 
+    protected AggregationBuilder(AggregationBuilder clone) {
+        this.name = clone.name;
```

Comment:
if this is supposed to produce and exact clone of the passed in builder then should we not set the factoriesBuilder here too?